### ISSUE TITLE
Integrate Brazilian Portuguese Plumi 4.0 Translation

### ISSUE THREAD

**Issue by mgogoulos** [plumi/plumi.app#1354]
**Reported by anna on 24 Jan 2011 06:21 UTC**

On 24/01/11 12:12 AM, Luciano De Fazio wrote:
> Hello, And.
>
> I am pleased to be able in send my small contribution for this great open source project.
>
> We are currently planning a migration from a multimedia management portal developed in Plone 2.5 to Plumi, for use by a muncipal council in Brazil.
>
> Once we complete the migration, we will be happy to send you the URL.
>
> Cheers,
>
> Luciano De Fazio
>
> 2011/1/22 And <andenator@gmail.com>
>
>     Hi Luciano,
>     that's fantastic. I'm cc'ing this to the main development list so people there can help you with the integration.
>
>     Also is it in use anywhere that we can see?
>
>     Thanks a bunch, very cool news.
>     And.
>

**Comment by mgogoulos** [comment]
**Modified by anna on 25 Jan 2011 01:21 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 26 Jan 2011 07:17 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 26 Jan 2011 07:17 UTC**
please implement this now if possible - let's be responsive to this contributor

**Comment by mgogoulos** [comment]
**Comment by dave on 26 Jan 2011 23:58 UTC**
I have set this up in plumi.locales, committed the .po file.

I am not so familiar with i18n process, following the README in the directory, I ran :

$ i18ndude sync --pot locales/plumi.pot locales/id/LC_MESSAGES/plumi.po locales/es/LC_MESSAGES/plumi.po locales/ms/LC_MESSAGES/plumi.po 

locales/pt_BR/LC_MESSAGES/plumi.po

locales/id/LC_MESSAGES/plumi.po: 0 added, 0 removed

locales/es/LC_MESSAGES/plumi.po: 0 added, 0 removed

locales/ms/LC_MESSAGES/plumi.po: 0 added, 0 removed

locales/pt_BR/LC_MESSAGES/plumi.po: 0 added, 0 removed

This seemed odd, so I tried regenerating the plumi.pot

$ i18ndude rebuild-pot --pot locales/plumi.pot --create plumi ../../../plumi.app/ ../../../plumi.content/ ../../../plumi.skin/

which resulted in a lot of errors, namely over the RSS2 areas (tal:content on empty tag etc)

After that, I synced again, this resulted in 6 added, 46 removed as above for each. Reading the actual file it looks ok, however it's probably good that someone familiar with internationalization confirm the process rather than me fuddle around with it at this stage. the '46 removed' has me concerned I have missed something.

**Comment by mgogoulos** [comment]
**Comment by anna on 27 Jan 2011 06:25 UTC**
hmm, i tried to test this, but it didn't seem to work. activating Indonesian didn't seem to work either...

http://testing.plumi.org:16080/Plone?set_language=id
http://testing.plumi.org:16080/Plone?set_language=pt-br

was this meant to be available on your testing site dave?

**Comment by mgogoulos** [comment]
**Comment by dave on 31 Jan 2011 07:45 UTC**
Just an update.

```
$ cd plumi.locales/plumi/locales

$ rm locales/plumi.pot

$ i18ndude rebuild-pot --pot locales/plumi.pot --create plumi ../../../plumi.app/ ../../../plumi.content/ ../../../plumi.skin/

```

This creates a whole string of warnings, I am not sure if they have to be fixed to resolve, or actually cause any issue or just warnings.

note: one is mentioned here : http://plumi.org/ticket/221

I used this workaround to test, but there are also outstanding ones with regard i18n tags in specifically the RSS feeds for news and events and callouts that require a custom tag, as the default is already defined elsewhere differently.

```
$ i18ndude sync --pot locales/plumi.pot locales/id/LC_MESSAGES/plumi.po locales/es/LC_MESSAGES/plumi.po locales/ms/LC_MESSAGES/plumi.po locales/pt_BR/LC_MESSAGES/plumi.po
```

Doing this on my http://localhost:8080/Plone?set_language=id or http://localhost:8080/Plone?set_language=pt-br I get no translation.

However :

http://testing.plumi.org:16080/Plone/Members/test/videos/test-video/?set_language=id

http://testing.plumi.org:16080/Plone/Members/test/videos/test-video/?set_language=pt-br

work.

I presume, but not familiar with translations, that I should fix all the issues with translation strings all through the code to fix this. I will go ahead with this, but noting where I am at in this ticket in case someone familiar with i18n can point me at the issue/process I am missing.

It appears that translations are half working, but only the plumi translations, not the plone ones.

On startup on my local installation I get warnings about the compiling:

```
2011-01-31T17:32:41 WARNING zope.i18n Error while compiling /opt/plumi/trunk/src/plumi.locales/plumi/locales/locales/id/LC_MESSAGES/plumi.po
------
2011-01-31T17:32:41 WARNING zope.i18n Error while compiling /opt/plumi/trunk/src/plumi.locales/plumi/locales/locales/es/LC_MESSAGES/plumi.po
------
2011-01-31T17:32:41 WARNING zope.i18n Error while compiling /opt/plumi/trunk/src/plumi.locales/plumi/locales/locales/ms/LC_MESSAGES/plumi.po
------
2011-01-31T17:32:41 WARNING zope.i18n Error while compiling /opt/plumi/trunk/src/plumi.locales/plumi/locales/locales/pt_BR/LC_MESSAGES/plumi.po
```

**Comment by mgogoulos** [comment]
**Modified by dave on 7 Feb 2011 04:00 UTC**

**Comment by mgogoulos** [comment]
**Comment by mike on 9 Feb 2011 18:10 UTC**
check it out on staging.engagemedia.org :)

**Comment by mgogoulos** [comment]
**Comment by and on 10 Feb 2011 02:51 UTC**
Awesome. Couple of things

* the flags are pushed across to the right of the page pushing the whole site out (firefox and chrome I have tested, may be others)
* you lose a lot of the pages in the left nav. when you choose pt-br. even if there are no translations the default should stay there
* looks like there is still a few things missing, but there is a lot in there, def. enough to ship with.

**Comment by mgogoulos** [comment]
**Comment by dave on 10 Feb 2011 03:09 UTC**
Hiya Mike, out of interest, what was missing in the process? Would it be a good idea to update the README.txt?

And, some of the things I have added in templates do not have i18n tags, being able to test the process for updating the files would give me the ability to add some of this for translation

**Comment by mgogoulos** [comment]
**Comment by and on 10 Feb 2011 12:55 UTC**
This is almost there so make it a blocker for rc1 release.

**Comment by mgogoulos** [comment]
**Comment by mike on 10 Feb 2011 20:19 UTC**
I've fixed the flags issue as far as I can tell (shift+refresh on staging).
The pages that we miss in the left nav are because of this linguaplone bug: http://plone.org/products/linguaplone/issues/262

dave, as far as I can tell, the README instructions still apply. But these are to create the .pot file and synchronize the .po files to the new pot. There are a couple of issues with linguaplone that you may have hit (like the tabs one described in the other ticket). Other than that, there was an issue with em.skin that prevented language changing that is now resolved.

**Comment by mgogoulos** [comment]
**Comment by and on 11 Feb 2011 05:47 UTC**
OK, so we make another ticket to follow that bug? Any ideas when it might be fixed? Can you make the ticket? You'll write it better than me ;)

Luciano does the translation look ok to you?

**Comment by mgogoulos** [comment]
**Comment by mike on 11 Feb 2011 09:50 UTC**
There is already that ticket on the linguaplone bug tracker which is in confirmed status. Usually linguaplone bugs get fixed soon. We should wait a few weeks and check the bug's progress.

**Comment by mgogoulos** [comment]
**Comment by and on 12 Feb 2011 07:10 UTC**
OK, I'll push this to 4.2 as something to check later on and remind us.

**Comment by mgogoulos** [comment]
**Modified by and on 14 Feb 2011 05:32 UTC**

**Comment by mgogoulos** [comment]
**Modified by and on 27 Feb 2011 07:19 UTC**

**Comment by mgogoulos** [comment]
**Comment by mike on 13 Mar 2011 10:01 UTC**
Brazilian translations are included on plumi 4.1 so I'm closing this and opening a new one about the tabs issue on 4.3
